### PR TITLE
refactor(react-grid): extract viewport getter for virtual table

### PR DESCRIPTION
**Pull request by ushkal**

### CODE REVIEW

File: packages/dx-grid-core/src/utils/virtual-table.ts [hunk 1]
```diff
@@ -80,7 +79,7 @@ export const getColumnBoundaries: PureComputed<
   )
 );
 export const getRowsVisibleBoundary: PureComputed<
-[TableRow[], number, number, GetColumnWidthFn, number, number, boolean?], RowsVisibleBoundary
+[TableRow[], number, number, GetColumnWidthFn, number, number, boolean?], VisibleBoundary
```
Comment (Krijovnick) [inline packages/dx-grid-core/src/utils/virtual-table.ts]: There is should be `GetRowHeightFn` type instead `GetColumnWidthFn`

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const rowsComputed = ({ virtualRows }: Getters) => plainRows(virtualRows);
 const loadedRowsStartComputed = ({ virtualRows }: Getters) => loadedRowsStart(virtualRows);
 
 // tslint:disable-next-line: max-line-length
-class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps, VirtualTableStateState> {
+class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps, any> {
```
Comment (Krijovnick) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: - The `VirtualTableStateState` type can be removed from `virtula-table-state.types.ts` file. 
- Should be there `any` state? Maybe `DeprecatedLazyLoadingStrategyState`?

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table.tsx [hunk 3]
```diff
@@ -24,7 +34,7 @@ export const makeVirtualTable: (...args: any) => any = (Table, {
   defaultEstimatedRowHeight,
   defaultHeight,
 }) => {
-  class VirtualTable extends React.PureComponent<VirtualTableProps, VirtualTableLayoutState> {
+  class VirtualTable extends React.PureComponent<VirtualTableProps, any> {
```
Comment (Krijovnick) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table.tsx]: There is also `any` state

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx (hunk 2 above)
Comment (ushkal) [reply packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: `VirtualTableStateState` is the one we need. `DeprecatedLazyLoadingStrategyState` is from another PR :)

**Pull request merged**